### DISCUSSION TITLE
Hopular networks : monitor for potential inclusion into tabular prediction

### DISCUSSION THREAD

**Discussion by willsmithorg** [autogluon/autogluon#1791] [Ideas]
This recently published paper and blog post claims that "Hopular networks" beat both gradient boosted trees and NN based approaches on small and medium sized tabular datasets.

The source code is not particular clear exactly what it does and how to use  on arbitrary tabular datasets, and this is definitely not drop-in-able yet as a model into autogluon tabular, but I'd say Hopular networks are well worth monitoring.

Code : https://github.com/ml-jku/hopular
Paper : https://arxiv.org/abs/2206.00664
Blog : https://ml-jku.github.io/hopular/

These claims from the authors are pretty attention grabbing:
-  _"In experiments on medium-sized tabular data with about 10,000 samples, Hopular outperforms XGBoost, CatBoost, LightGBM and a state-of-the art Deep Learning method designed for tabular data."_
- _"We show that the attention mechanism of transformer architectures is actually the update rule of modern Hopfield networks that can store exponentially many patterns."_

**Comment by sxjscience** [comment]
Yeah, the paper is also on our radar. Actually, we are taking a look at advanced Tabular DL models, e.g., via https://github.com/awslabs/autogluon/pull/1655 and integrating [FT-Transformer](https://arxiv.org/pdf/2106.11959.pdf) to AutoMM: https://github.com/awslabs/autogluon/tree/master/examples/automm

**Comment by Innixma** [comment]
Thanks for mentioning this new model + code @willsmithorg!

I am always wary of papers that claim NN are beating GBMs in Tabular data. In their paper, they use a small set of datasets and don't seem to early stop the GBM models and only fit exactly 1000 iterations (often not enough). They also don't mention if they take the best iteration on val data or if they just take iteration 1000 as the final model. This is extra suspect because they hyperparameter tune their model with up to 10,000 epochs and early stopping. Curious.

Looking into it a bit deeper, seems I am not alone in my concerns: https://medium.com/@tunguz/trouble-with-hopular-6649f22fa2d3

Regardless, it is still probably worth looking into at some point to see if their claims hold up in practice if someone is willing to perform the code integration into AG.
